tls/s2n_key_update.c

```diff
@@ -53,6 +53,10 @@ int s2n_key_update_send(struct s2n_connection *conn, s2n_blocked_status *blocked
     POSIX_ENSURE_REF(conn);
     POSIX_ENSURE_REF(conn->secure);
 
+    if (conn->actual_protocol_version < S2N_TLS13) {
```

Comment:
Should we do this for all post handshake messages?

Reply:
We already do it for session tickets: https://github.com/aws/s2n-tls/blob/main/tls/s2n_server_new_session_ticket.c#L124-L126

But I can move the check up to s2n_post_handshake_send. We can't do the same with s2n_post_handshake_recv though, so it ruins the symmetry a bit.